### DISCUSSION TITLE
Github actions examples

### DISCUSSION THREAD

**Discussion by razzeee** [tree-sitter/tree-sitter#817] [Libraries & tools]
We've switched our tree sitter implementation over to github actions and I thought it might be useful for others, that are still on travis/appveyor.

https://github.com/Razzeee/tree-sitter-elm/tree/master/.github/workflows

There are pipelines for:
- Automatical Npm publish (gpr didn't work for some reason, so I omitted it)
- Prebuilds for linux and windows (I had problems with macos)
- Automatical publish for the rust crate
- Basic tree sitter tests and some hand chosen git repos that get parsed + very simple rust tests
- Parsing all elm sources we can find on github

Hope this is helpful for some of you

**Comment by Tuxified** [comment]
Nice, I was actually looking for ways to publish to npm/crates.io. If anyone is interested, the project I'm working on also publishes a playground to github pages on every green merge to main: https://github.com/Tuxified/tree-sitter-elixir/tree/main/.github/workflows

**Comment by razzeee** [comment]
GPR publish now seems to work, not sure what the initial problem was

**Comment by ObserverOfTime** [comment]
See https://github.com/tree-sitter/workflows
